apps/api/R/submit.workflow.R

```diff
@@ -0,0 +1,180 @@
+library(dplyr)
+
+#* Submit a workflow submitted as XML
+#* @param workflowXmlString String containing the XML workflow from request body
+#* @param userDetails List containing userid & username
+#* @return ID & status of the submitted workflow
+#* @author Tezan Sahu
+submit.workflow.xml <- function(workflowXmlString, userDetails){
+  
+  workflowXml <- XML::xmlParseString(stringr::str_replace(workflowXmlString, "<?.*?>\n", ""))
+  workflowList <- XML::xmlToList(workflowXml)
+  
+  # Fix details about the database
+  workflowList$database <- list(bety = PEcAn.DB::get_postgres_envvars(
+      host = "localhost",
+      dbname = "bety",
+      user = "bety",
+      password = "bety", 
+      driver = "PostgreSQL"
+    )
+  )
+  
+  # Fix RabbitMQ details
+  dbcon <- PEcAn.DB::betyConnect()
+  hostInfo <- PEcAn.DB::dbHostInfo(dbcon)
+  PEcAn.DB::db.close(dbcon)
+  workflowList$host <- list(
+    rabbitmq = list(
+      uri = Sys.getenv("RABBITMQ_URI", "amqp://guest:guest@localhost/%2F"),
+      queue = paste0(workflowList$model$type, "_", workflowList$model$revision)
+    )
+  )
+  workflowList$host$name <- if(hostInfo$hostname == "") "localhost" else hostInfo$hostname
+  
+  # Fix the info
+  workflowList$info$notes <- workflowList$info$notes
+  if(is.null(workflowList$info$userid)){
+    workflowList$info$userid <- userDetails$userid
+  }
+  if(is.null(workflowList$info$username)){
+    workflowList$info$username <- userDetails$username
+  }
+  if(is.null(workflowList$info$date)){
+    workflowList$info$date <- Sys.time()
+  }
+  
+  # Add entry to workflows table in database
+  workflow_id <- insert.workflow(workflowList)
+  workflowList$workflow$id <- workflow_id
+  
+  # Add entry to attributes table in database
+  insert.attribute(workflowList)
+  
+  # Fix the output directory
+  outdir <- paste0("/data/workflows/PEcAn_", workflow_id)
```

Comment:
I understand not wanting to let the external user to specify the outdir in their XML (security nightmare and external user doesn't need/want to know where files are stored), but this hardcoded path is going to end up putting PEcAn output in places the host system doesn't want and can't control. e.g. on my pecan server all the runs go to a specific disk mount, the defaults for which are currently stored in the config.php (and we've written R functions that other apps [e.g. Shiny] are using to access info in that config).

Reply:
I had a discussion with @robkooper about this and we mutually came to a conclusion to go ahead in this way. Maybe I'll talk to him again about how we could modify it if needed.

Reply:
Yeah, I think it would be really helpful for the host to be able to (optionally) set system-specific configurations like the root dirs for inputs, outputs, etc.

Reply:
I have created an env var to sort this after discussing with @robkooper